src/vs/editor/contrib/codeAction/browser/codeActionModel.ts

```diff
@@ -197,12 +205,82 @@ export class CodeActionModel extends Disposable {
 					return;
 				}
 
-				const actions = createCancelablePromise(token => getCodeActions(this._registry, model, trigger.selection, trigger.trigger, Progress.None, token));
+				const startPosition = trigger.selection.getStartPosition();
+
+				const actions = createCancelablePromise(async token => {
+					if (this._settingEnabledNearbyQuickfixes() && trigger.trigger.type === CodeActionTriggerType.Invoke && trigger.trigger.triggerAction === CodeActionTriggerSource.QuickFix) {
```

Comment:
For future proofing, I still think it would be better to check `trigger.filter` instead of using `trigger.trigger.triggerAction === CodeActionTriggerSource.QuickFix`. That way if someone setup a custom keybinding for a specific code action, this logic will kick in

Reply:
The problem with using filter is that it is basically always undefined/null. `trigger.trigger.triggerAction === CodeActionTriggerSource.QuickFix` would still cover individuals with custom keybindings.

Tracing from here:
https://github.com/microsoft/vscode/blob/31067371f2aee12f5dc0a8e316141af77f2870fe/src/vs/editor/contrib/codeAction/browser/codeActionCommands.ts#L83

Reply:
Users can create custom keybindings for `editor.action.codeAction` which instead uses `CodeActionTriggerSource.Default`. `CodeActionTriggerSource` is really only meant to be used for telemetry purposes